### ISSUE TITLE
imread() for tiff file returns wrong dimensions

### ISSUE THREAD

**Issue by r0f1** [scikit-image/scikit-image#5545]
Consider this following code:

```python
from pathlib import Path
from skimage.io import imread

a = sorted(Path(input_dir).glob("*.tiff"))[0]

imread(a).shape
imread(str(a)).shape
```
Output:
```
(64, 6)     # wrong
(64, 64, 6) # correct
```
I am trying to read a tiff file with 6 color channels. Why do I get different image dimension depending on whether I pass a PosixPath or a string to `imread()`? If PosixPaths are not supported by `imread()`, which is totally fine, could you please throw a ValueError?

**Comment by r0f1** [comment]
Thanks, that's awesome!

**Comment by oreo2107** [comment]
Is this issue still open for a fix?

**Comment by grlee77** [comment]
@oreo2107, this one has a fix already proposed in #5546
